frontend/src/metabase/components/SelectList/SelectList.tsx

```diff
@@ -1,14 +1,17 @@
-import type * as React from "react";
-import { forwardRef } from "react";
+import { type HTMLProps, forwardRef } from "react";
+
+import { Box, type BoxProps } from "metabase/ui";
 
 import { BaseSelectListItem } from "./BaseSelectListItem";
 import { SelectListItem } from "./SelectListItem";
 
-type SelectListProps = Omit<React.HTMLProps<HTMLUListElement>, "role">;
+type SelectListProps = Omit<HTMLProps<HTMLUListElement>, "role"> & BoxProps;
 
 const SelectList = forwardRef<HTMLUListElement, SelectListProps>(
   function SelectList(props: SelectListProps, ref) {
-    return <ul {...props} ref={ref} role="menu" data-testid="select-list" />;
+    return (
+      <Box<"ul"> {...props} ref={ref} role="menu" data-testid="select-list" />
```

Comment:
I thought `Box<"ul">` only casts the type to "ul" and didn't actually use the `ul` tag under the hood. Should we use the `component` prop instead per [their docs](https://v6.mantine.dev/core/box/#polymorphic-component)?


```suggestion
      <Box {...props} ref={ref} role="menu" data-testid="select-list" component="ul" />
```

Reply:
yes definitely 😄